src/mesh/mesh_function.C

```diff
@@ -647,6 +647,45 @@ const Elem * MeshFunction::find_element(const Point & p,
   // locate the point in the other mesh
   const Elem * element = (*_point_locator)(p, subdomain_ids);
 
+  // Make sure that the element found is evaluable
+  return this->check_found_elem(element, p);
+}
+
+std::set<const Elem *> MeshFunction::find_elements(const Point & p,
+                                                   const std::set<subdomain_id_type> * subdomain_ids) const
+{
+  // Ensure that in the case of a master mesh function, the
+  // out-of-mesh mode is enabled either for both or for none.  This is
+  // important because the out-of-mesh mode is also communicated to
+  // the point locator.  Since this is time consuming, enable it only
+  // in debug mode.
+#ifdef DEBUG
+  if (this->_master != nullptr)
+    {
+      const MeshFunction * master =
+        cast_ptr<const MeshFunction *>(this->_master);
+      libmesh_error_msg_if(_out_of_mesh_mode!=master->_out_of_mesh_mode,
+                           "ERROR: If you use out-of-mesh-mode in connection with master mesh "
+                           "functions, you must enable out-of-mesh mode for both the master and the slave mesh function.");
+    }
+#endif
+
+  // locate the point in the other mesh
+  std::set<const Elem *> candidate_elements;
+  std::set<const Elem *> final_candidate_elements;
+  (*_point_locator)(p, candidate_elements, subdomain_ids);
+
+  // For each candidate Elem, if it is evaluable, add it to the set of
+  // final candidate Elems.
+  for (const auto & element : candidate_elements)
+    final_candidate_elements.insert(this->check_found_elem(element, p));
```

Comment:
We should be filtering out `nullptr` return values (which `check_found_elem` can return) at some point, right?  Since this is a `std::set` I'd just `erase(nullptr)` after the loop here.

Reply:
Nope, never mind.  We were already inserting `nullptr` in that else clause.  This is stupid and we should change it, but probably not in this PR and probably not until we've had some time to make sure users don't report any fallout from this PR that our test suites missed.